VideoLocker/src/main/java/org/edx/mobile/http/OfflineRequestInterceptor.java

```diff
@@ -27,16 +31,18 @@ public OfflineRequestInterceptor(Context context, int maxStaleTime){
     }
 
     @Override
-    public void intercept(RequestInterceptor.RequestFacade request) {
+    public Response intercept(Chain chain) throws IOException {
+        final Request.Builder requestBuilder = chain.request().newBuilder();
         if (isNetworkConnected()) {
             //TODO? should we specify caching here?
```

Comment:
From the commented implementation below, it looks like this is referring to reading from recently cached data even while online. I don't think this is part of our requirements anywhere, but the proper way to do this would probably be to use the [`stale-while-revalidate`][stale-while-revalidate] header. Anyway, it's out of scope here, but might be revisited in [MA-2412][]. What do you think, @bguertin, @BenjiLee, @mdinino?

[stale-while-revalidate]: https://tools.ietf.org/html/rfc5861
[MA-2412]: https://openedx.atlassian.net/browse/MA-2412